relations/src/gr1cs/constraint_system.rs

```diff
@@ -0,0 +1,773 @@
+//! This module contains the implementation of the `ConstraintSystem` struct.
+//! a constraint system contains multiple predicate constraint systems,
+//! each of which enforce have seperate predicates and constraints. For more infomation about the terminology and the structure of the constraint system, refer to https://eprint.iacr.org/2024/1245
```

Comment:
Maybe add a specific section in the paper?
```suggestion
//! each of which enforce have seperate predicates and constraints. 
//! For more infomation about the terminology and the structure of the constraint system, refer to https://eprint.iacr.org/2024/1245.
```